### PR TITLE
fix: updating the Docker Repository credential configuration

### PR DESCRIPTION
**Pull request by hferentschik**
Updating management of `jenkins-docker-cfg` Secret.

Superseding pull request #2619.

### CODE REVIEW

**Comment by hferentschik** [comment]
@daveconde - I've just seen the issue you created - https://github.com/jenkins-x/jx-docs/issues/2600 - and wondering what you think about this pull request. I am a bit confused about you mentioning the requirements file. Changing the registry is one thing, but this particular page is about changing the credentials for the registry which is still valid or am I missing something?

File: content/en/docs/reference/components/docker-registry.md [hunk 1]
```diff
@@ -8,99 +8,93 @@ aliases:
...
+## Creating `config.json`
 
-## Using a different Docker Registry
+### Google Container Registry (GCR)
 
-If you are using the public cloud you may wish to take advantage of your cloud providers docker registry; or reuse your own existing docker registry.
-
-### If you are using Static Jenkins Master
-To specify the Docker Registry host/port you can use the Jenkins Console:
+If you wany to use GCR you can create your `config.json` by running:
```
Comment (hervelemeur) [inline content/en/docs/reference/components/docker-registry.md:18]:
```suggestion
If you want to use GCR you can create your `config.json` by running:
```

File: content/en/docs/reference/components/docker-registry.md [hunk 2]
```diff
@@ -8,99 +8,93 @@ aliases:
 ---
 
 To be able to create and publish docker images we need to use a Docker Registry.
+If you want to change the default registry you will need to configure and update the `jenkins-docker-cfg` Kuberenetes Secret in the Jenkins X system namespace.
```
Comment (hervelemeur) [inline content/en/docs/reference/components/docker-registry.md:11]:
```suggestion
If you want to change the default registry you will need to configure and update the `jenkins-docker-cfg` Kubernetes Secret in the Jenkins X system namespace (named by default `jx`).
```

File: content/en/docs/reference/components/docker-registry.md [hunk 3]
```diff
@@ -8,99 +8,93 @@ aliases:
...
+docker logout
+docker login -u <username> -p <password>
 ```
-2. Create a local file called `config.json` and its value should be in this format (update values based on your registry user account and FQDN).
+
+Once you have your custom `config.json` refer to [Updating the config.json secret](/docs/reference/components/docker-registry/#updating-the-jenkins-docker-cfg-secret) to see how to update the `jenkins-docker-cfg` Secret.
+
+### jFrog BinTray (Artifactory)
+
+Using the jFrog BinTray as a private registry is possible.  This has only been tested when creating a new cluster and passing the `--docker-registry=private-reg.bintray.io`.
```
Comment (hervelemeur) [inline content/en/docs/reference/components/docker-registry.md:96]:
```suggestion
Using the jFrog BinTray as a private registry is possible. This has only been tested when creating a new cluster and passing the `--docker-registry=private-reg.bintray.io`.
```

File: content/en/docs/reference/components/docker-registry.md [hunk 4]
```diff
@@ -112,16 +106,16 @@ kubectl delete secret jenkins-docker-cfg
...
-### Mount a Secret for your registry
+Provided you have your custom `config.json`, you can update the `jenkins-docker-cfg` secret by running:
 
-Your docker registry will require a Secret to be mounted into the [Pod Templates](/docs/managing-jx/common-tasks/pod-templates/).
+```sh
+kubectl delete secret jenkins-docker-cfg -n jx
+kubectl create secret generic jenkins-docker-cfg -n jx --from-file=./config.json
+```
 
+**NOTE**: the file must be called `config.json` as the file name is used in the key of the underlying `Secret` in kubernetes
```
Comment (hervelemeur) [inline content/en/docs/reference/components/docker-registry.md:121]:
```suggestion
**NOTE**: the file must be called `config.json` as the file name is used in the key of the underlying `Secret` in kubernetes.
```

**Comment by hferentschik** [comment]
@hervelemeur for the review :-)

**Comment by hferentschik** [comment]
http://jx-docs.jx-jenkins-x-jx-docs-pr-2645.jenkins-x.live/docs/reference/components/docker-registry/

File: content/en/docs/reference/components/docker-registry.md [hunk 5]
```diff
@@ -7,100 +7,206 @@ aliases:
...
-You can do this via the [jx create docker auth](/commands/jx_create_docker/) command line tool:
+{{% alert %}}
+An alternative approach is to just set `enableDocker: true` and run `jx boot` locally.
+In this case, it will interactively ask for the required parameters again and persist them into `env/parameters.yaml` and the underlying secret store.
+{{% /alert  %}}
 
+## Update secret store
+
+The next step is to make sure the password is stored in the secret store.
+Assuming you are using [Vault](/docs/install-setup/installing/boot/secrets/#vault) as secret store, you need to make sure the secret identified by the URI _vault:\<cluster-name\>/docker:password_ exists.
```
Comment (jha-cloudbees) [inline content/en/docs/reference/components/docker-registry.md:68]:
```suggestion
Assuming you are using [Vault](/docs/install-setup/installing/boot/secrets/#vault) as the secret store, you need to make sure the secret identified by the URI _vault:\<cluster-name\>/docker:password_ exists.
```

File: content/en/docs/reference/components/docker-registry.md [hunk 6]
```diff
@@ -7,100 +7,206 @@ aliases:
...
+This can be achieved by running (you need the `vault` CLI installed for that):
+
+```sh
+eval $(jx get vault-config)
+vault kv put /secret/<cluster-name>/docker passwor=<my-password>
 ```
-jx create docker auth --host "foo.private.docker.registry" --user "foo" --secret "FooDockerHubToken" --email "fakeemail@gmail.com"
 
+You find more information on how to interact with your secrets in Vault in the [Manage your secrets](/docs/reference/components/vault/) section.
```
Comment (jha-cloudbees) [inline content/en/docs/reference/components/docker-registry.md:76]:
```suggestion
You can find more information on how to interact with Vault secrets in the [Manage your secrets](/docs/reference/components/vault/) section.
```

File: content/en/docs/reference/components/docker-registry.md [hunk 7]
```diff
@@ -7,100 +7,206 @@ aliases:
...
+
+## Update Kubernetes provider configuration
+
+Last but not least, you need to make sure that the correct Docker authentication `config.json` gets generated and stored in the Kubernetes Secret `jenkins-docker-cfg` (within your development namespace).
```
Comment (jha-cloudbees) [inline content/en/docs/reference/components/docker-registry.md:80]:
```suggestion
Finally, you need to make sure that the correct Docker authentication `config.json` gets generated and stored in the Kubernetes Secret `jenkins-docker-cfg` (within your development namespace).
```

File: content/en/docs/reference/components/docker-registry.md [hunk 8]
```diff
@@ -7,100 +7,206 @@ aliases:
...
+    "credsStore": "ecr-login"
+}
+```
 
-## Using Docker Hub
+### Docker Hub
 
-If you want to publish images to docker hub then you need to modify your `config.json` as described above to something like:
+If you want to publish images to Docker Hub, then you need a `config.json` with and _auth_ section containing your Docker Hub auth token.
+Something like:
```
Comment (jha-cloudbees) [inline content/en/docs/reference/components/docker-registry.md:171]:
```suggestion
For example:
```

File: content/en/docs/reference/components/docker-registry.md [hunk 9]
```diff
@@ -7,100 +7,206 @@ aliases:
...
 ```
-2. Create a local file called `config.json` and its value should be in this format (update values based on your registry user account and FQDN).
+
+On macOS you might find something like this:
+
+```json
+"credsStore": "osxkeychain"
+```
+
+in `.docker/config.json` without and _auths_ section.
```
Comment (jha-cloudbees) [inline content/en/docs/reference/components/docker-registry.md:197]:
```suggestion
in `.docker/config.json` without an _auths_ section.
```

**Comment by jha-cloudbees** [comment]
just some fixes, but awesome docs work!

**Comment by hferentschik** [comment]
@jha-cloudbees - all fixes applied, thanks

**Comment by hferentschik** [comment]
@dgozalo did you get a chance to look at this?

**Pull request closed without merging**